src/lib.rs

```diff
@@ -279,6 +282,14 @@ fn gracefully_shut_down_indexer() {
       log::warn!("Index thread panicked; join failed");
     }
   }
+
+  if let Some(search_indexer) = SEARCH_INDEXER.lock().unwrap().take() {
+    shut_down();
+    log::info!("Waiting for search index thread to finish...");
+    if search_indexer.join().is_err() {
+      log::warn!("Search index thread panicked; join failed");
+    }
```

Comment:
Just good for debuggability:

```suggestion
    if let Err(err) = search_indexer.join() {
      log::warn!("Search index thread panicked: {err}");
    }
```